Various small fixes and web-client features

sisou opened the pull request:
## What's in this pull request?

Various fixes for small issues I found during work on #<next PR>, plus two small features for the web-client.
See commits for a list.

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

sisou committed:
web-client: Add Hash class with common hashing methods

sisou committed:
web-client: Add constructor to Address

sisou committed:
Fix error arguments for Transaction deserialization

sisou committed:
Small cleanup of unnecessary ref and duplicated method

sisou committed:
Fix web-client KeyPair.signTransaction method for basic transactions

sisou committed:
Fix web-client ClientConfiguration.network() rustdoc

jsdanielh commented on web-client/src/primitives/hash.rs:
```diff
@@ -0,0 +1,29 @@
+#[wasm_bindgen]
+impl Hash {
+    #[wasm_bindgen(js_name = computeBlake2b)]
+    pub fn compute_blake2b(data: &[u8]) -> Vec<u8> {
```
Could you add some small `rustdoc`?

sisou replied:
Do we really need a rustdoc? The method name is pretty self-explanatory?

sisou replied:
> Computes a 32-byte Blake2b hash from the input data. Blake2b is used for example to compute a public key's address.

jsdanielh commented on web-client/src/primitives/hash.rs:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    #[wasm_bindgen(js_name = computeArgon2d)]
+    pub fn compute_argon2d(data: &[u8]) -> Vec<u8> {
```
Could you add some small `rustdoc`?

sisou replied:
> Computes a 32-byte Argon2 hash from the input data. Argon2d hashes are used for example for [purpose].

Please find out for what, @jsdanielh 😅

jsdanielh replied:
We actually don't have a specific usage for it :smile:

sisou replied:
I think it was the PoW hash.

jsdanielh replied:
You are right

jsdanielh commented on web-client/src/primitives/hash.rs:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    #[wasm_bindgen(js_name = computeSha256)]
+    pub fn compute_sha256(data: &[u8]) -> Vec<u8> {
```
Could you add some small `rustdoc`?

sisou replied:
> Computes a 32-byte SHA256 hash from the input data. SHA256 is a common hash algorithm.

jsdanielh commented on web-client/src/primitives/hash.rs:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    #[wasm_bindgen(js_name = computeSha512)]
+    pub fn compute_sha512(data: &[u8]) -> Vec<u8> {
```
Could you add some small `rustdoc`?

sisou replied:
Haha, I'm tempted to just put `/// small` now everywhere 😅

I don't have my laptop/SSH key now, but I'll formulate the rustdoc here and then you can add it, ok?

sisou replied:
> Computes a 64-byte SHA512 hash from the input data.

jsdanielh replied:
> I don't have my laptop/SSH key now, but I'll formulate the rustdoc here and then you can add it, ok?

Yup, no problem

jsdanielh committed:
web-client: Fix clippy warning

Fix a clippy warning the `web-client` related to using `clone` on a
type that implements `Copy`.
